src/lib/Gitolite/Conf/Load.pm

```diff
@@ -521,9 +521,10 @@ sub user_roles {
     # - else give everyone read access to everything
     my @retval = ( '@READERS' );
 
-    # - and give owners '@CREATOR' access to their repos
+    # - and give owners '@CREATOR' access to their repos 
+    # - http_anonymous user should never receive a CREATOR role
     my ( $owner, $project ) = split('/', $repo, 2);
-    if ( $owner eq $user ) {
+    if ( $owner eq $user and $user ne 'http_anonymous') {
```

Comment:
I decided to extend the condition in the existing if statement. Another approach I considered was adding a separate block that checks for the anonymous user and returns the existing @retval which only contains the @READERS value when the user matches.
```
    if ( $user eq 'http_anonymous') {
        return @retval;
    }
```
I think it's more explicit, however I realized that returning early would also skip the debugging code at the end of the function (L530). So instead I modified the existing statement instead and kept the flow as is with only one return statement at the end of the function. But I would be happy to get feedback.